server/src/test/java/server/formatting/FormattingProviderTest.java

```diff
@@ -0,0 +1,144 @@
+package server.formatting;
+
+import java.io.File;
+import java.util.concurrent.CompletableFuture;
+import java.util.List;
+
+import org.eclipse.lsp4j.DocumentFormattingParams;
+import org.eclipse.lsp4j.FormattingOptions;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.TextEdit;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import server.buildifier.Buildifier;
+import server.buildifier.BuildifierException;
+import server.buildifier.BuildifierFileType;
+import server.buildifier.FormatInput;
+import server.buildifier.FormatOutput;
+import server.utils.DocumentTracker;
+
+public class FormattingProviderTest {
+
+    
+    FormattingProvider formattingProvider;
+
+    DocumentTracker documentTracker;
+    Buildifier buildifier;
+    File file;
+    DocumentFormattingParams params;
+
+    @Captor
+    ArgumentCaptor<FormatInput> captor;
+
+    CompletableFuture<List<? extends TextEdit>> output;
+
+    @Before
+    public void setUp() throws Exception {
+        buildifier = Mockito.mock(Buildifier.class);
+        documentTracker = Mockito.mock(DocumentTracker.class);
+        FormattingProvider concreteFormattingProvider = new FormattingProvider(documentTracker, buildifier);
+        formattingProvider = Mockito.spy(concreteFormattingProvider);
+        file = Mockito.mock(File.class);
+        params = new DocumentFormattingParams(new TextDocumentIdentifier("DummyUriString"), new FormattingOptions());
+        captor = ArgumentCaptor.forClass(FormatInput.class);
+        output = null;
+
+        Mockito.doReturn(file).when(formattingProvider).getFileFromUriString(Mockito.any());
+        Mockito.when(file.getName()).thenReturn("BUILD");
+        Mockito.when(file.toURI()).thenReturn(null);
+        Mockito.when(documentTracker.getContents(Mockito.any())).thenReturn("Dummy Contents");
+        Mockito.when(buildifier.format(Mockito.any())).thenReturn(new FormatOutput("Dummy Output"));
+    }
+
+    @Test
+    public void FormattingProviderReturnsBuildifierOutputAsResult() throws Exception {
```

Comment:
Uppercase function names inconsistent with rest of code base

Reply:
Ok I updated those. Thanks.